### PR TITLE
doc(decisions): adopt DESIGN.md spec for repo-local design guides [KB-61]

### PR DESCRIPTION
**Pull request by vidbina**
## Summary

- Survey existing standards for repo-local design guides (DESIGN.md spec, awesome-design-md, GitLab Pajamas, UNICEF, MetaBrainz, contribute-design)
- Document decision to adopt Google's DESIGN.md open spec — keep the file spec-pure, route Figma workflow to a dedicated skill, route agent instructions to AGENTS.md
- Define project-specific vs KB-scoped boundary for design guide content
- Add spec-compliant `templates/DESIGN.md` with PROJECT-SPECIFIC/CROSS-PROJECT markers
- Update AGENTS.md template, README, and PROJECT_SETUP_GUIDE.md to reference DESIGN.md
- Filed KB-62 for Figma conventions skill extraction

## Test plan

- [x] Decision record follows template format (frontmatter, status log, all sections)
- [x] `templates/DESIGN.md` sections follow spec-prescribed order
- [ ] Links in decision record and AGENTS.md resolve to real pages
- [x] PROJECT_SETUP_GUIDE.md setup steps are actionable

Closes KB-61

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/decisions/0003-design-guide-standards.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+**Adopt the [DESIGN.md open spec](https://github.com/google-labs-code/design.md/blob/main/docs/spec.md) (Alternative B) for repo-local design guides. Keep the file spec-pure. Route non-spec concerns to their proper homes.**
+
+Specifically:
+
+1. **`DESIGN.md`** follows the community standard naming and section structure. Project-specific sections (Overview, Colors, Typography, Components, Do's & Don'ts) are marked with HTML comments (`<!-- PROJECT-SPECIFIC -->`) in the KB template so agents and humans can identify what needs customization per project.
+
+2. **AGENTS.md** (the template, not the KB's own) references DESIGN.md as required reading when design work starts. It also points to Figma workflow guidance and the design onboarding path — described by function, not by tool-specific endpoint names, so the guidance works across different agent systems.
+
+3. **Figma workflow patterns** (edit the base component, variables-first, verify visually, friction log of Plugin API edge cases) live in a dedicated capability separate from DESIGN.md. This is semantically distinct: DESIGN.md = "what to design", the Figma capability = "how to work effectively in Figma."
```
Comment (vidbina) [inline docs/decisions/0003-design-guide-standards.md]: doesnt say where exactly. will this be a skill? i think we landed on that with [KB-62](https://linear.app/asabina/issue/KB-62/create-figma-conventions-skill-from-ivos-trades-design-guide-patterns)?

File: docs/decisions/0003-design-guide-standards.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+- Ecosystem compatibility: CLI validation, Figma plugin generation, awesome-design-md examples as starting points
+- Zero spec maintenance burden — Google Labs and the community maintain the standard
+- Clean separation between design specification (DESIGN.md), tooling workflow (Figma conventions), and agent instructions (AGENTS.md)
+- AI agents with DESIGN.md awareness can consume our files natively
+
+**Negative:**
+
+- Our ICP framing must fit within the spec's Overview section — no dedicated section for it
+- Figma-specific guidance is split across two locations (general design principles in DESIGN.md, tool workflow in the Figma conventions capability)
+- Projects that don't use Figma still get a template that assumes the Figma ecosystem (plugin, variable export) — may need adaptation for other tools
```
Comment (vidbina) [inline docs/decisions/0003-design-guide-standards.md]: how? aren't we keeping that out with [KB-62](https://linear.app/asabina/issue/KB-62/create-figma-conventions-skill-from-ivos-trades-design-guide-patterns)?

File: templates/DESIGN.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+# Validate with: npx @google/design.md validate DESIGN.md
+# Generate from Figma: https://github.com/bergside/design-md-figma
+#
+# Sections marked <!-- PROJECT-SPECIFIC --> must be authored per project.
+# Sections marked <!-- CROSS-PROJECT --> carry reusable guidance — customize
+# values but keep the structure.
+
+name: "Project Name"
+
+# <!-- PROJECT-SPECIFIC: replace with your project's color tokens -->
```
Comment (vidbina) [inline templates/DESIGN.md:19]: is tokens spec-specific language, or have we decided to call it this?. Just asking because tokens initially led me to think it had anything to do with a scheme for token-optimised consumption by AIs but I think here it is just used as a spec-local alt for YAML node effectively.

File: templates/DESIGN.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+- Never introduce intermediate sizes outside the defined scale
+- Same-level items differentiate through weight and color, not size
+- All text nodes must reference a token — never hardcode size/weight
+- Components that serve the same role must use the same text style
+
+<!-- PROJECT-SPECIFIC -->
+
+## Layout & Spacing
+
+Spacing follows an 8px base unit scale. Consistent spacing rhythm creates
```
Comment (vidbina) [inline templates/DESIGN.md]: Crazy specific. Is this really how the spec works with this? And are absolute values preferred over relative values, in which case it would help to make that clear as well (unless already stated somewhere in this doc).

File: templates/DESIGN.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+- **full** (9999px) — pills, avatars, circular elements
+
+[Adjust the scale and mapping to match your project's visual language.]
+
+<!-- CROSS-PROJECT -->
+
+## Components
+
+[Document component-specific styling guidance here. Use sub-headings for
+component categories. Focus on token usage, interactive states, and structural
```
Comment (vidbina) [inline templates/DESIGN.md]: again… here token-usage is not LLM-related, i assume… or is it? Language just confuses me a bit so perhaps we can preamble this better.

File: templates/DESIGN.md [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+  clip content silently
+
+<!-- PROJECT-SPECIFIC -->
+
+## Do's and Don'ts
+
+**Do:**
+
+- Verify both light and dark modes after every visual change
+- Test with varying data lengths (short names, long names, edge cases)
```
Comment (vidbina) [inline templates/DESIGN.md:271]: where this makes sense. Some items will have fixed or capped data lengths in which case, the different length checks have known bounds and if these bounds are narrow enough, it may not be worthwhile to spend AI reasoning effort on this to just work it through for busy-work's sake. At least not from the very beginning, but it would be helpful as a post-prototyping refinement pass (to keep design docs complete). When we're moving fast, we need to make sure our tooling doesn't spent too much time on following a framework to the t on concerns that aren't most prudent at the time (we're under time pressure) but arguably this is not a DESIGN.md concern but rather something that fits into our design skill ([KB-62](https://linear.app/asabina/issue/KB-62/create-figma-conventions-skill-from-ivos-trades-design-guide-patterns)) or in our agent memory. Curious to input here.

File: PROJECT_SETUP_GUIDE.md [hunk 7]
```diff
@@ -196,6 +196,42 @@ cp -r <path-to-kb>/templates/decisions ./docs/
...
+
+If the project has a visual design component (UI, components, Figma integration), set up a `DESIGN.md` file. This follows the [DESIGN.md open spec](https://github.com/google-labs-code/design.md/blob/main/docs/spec.md).
+
+**Option A: Seed from an existing Figma file**
+
+If the project already has a Figma file with defined styles and variables, use the [Figma-to-DESIGN.md plugin](https://github.com/bergside/design-md-figma) to generate a starter file:
+
+1. Install the plugin in Figma
+2. Run it on your project file — it extracts local styles into DESIGN.md format
+3. Review and customize the generated file
```
Comment (vidbina) [inline PROJECT_SETUP_GUIDE.md]: before this we should present the option to expand the file or align the file with what we have in our templates/DESIGN.md just to make sure we close the gap. I'm not sure what the DESIGN.md that Figma produces covers (in particularly which parts it will lack compared to our template).

File: templates/AGENTS.md [hunk 8]
```diff
@@ -65,6 +65,17 @@ npm run lint --help && npm run test --help && npm run build --help
...
+2. **Figma workflow conventions** — if the project uses Figma, consult the team's Figma workflow guidance for patterns on variable architecture, component editing discipline (always edit the base component, not instances), and visual verification after changes.
+
+If `DESIGN.md` does not exist yet, consult the [KB template](https://github.com/google-labs-code/design.md/blob/main/docs/spec.md) and the project lead before creating one. A DESIGN.md can be seeded from an existing Figma file using the [Figma-to-DESIGN.md plugin](https://github.com/bergside/design-md-figma) or authored manually following the spec.
+
+Sections marked `<!-- PROJECT-SPECIFIC -->` must be authored per project (brand personality, color palette, ICP framing, component inventory). Sections marked `<!-- CROSS-PROJECT -->` carry reusable structure — customize values but keep the structure.
```
Comment (vidbina) [inline templates/AGENTS.md:77]: we should make it clear where things are unclear during design and warrant input from the user that we should flag it. Even if a decision is make to quickly solve a problem by making a decisions for the user, it needs to be surfaced as something that may need to be codified into the DESIGN.md for completeness sake. This way we keep the DESIGN.md up-to-date. We should also proactively raise conflicts between what is in design files and what we have in DESIGN.md in order to help the operators keep these in sync.

**Review by vidbina — COMMENTED** [review]
lgtm
